chaste_codegen/rl_model.py

```diff
@@ -0,0 +1,139 @@
+import time
+
+import sympy as sp
+
+import chaste_codegen as cg
+from chaste_codegen._linearity_check import derives_eqs_partial_eval_non_linear, get_non_linear_state_vars
+from chaste_codegen._partial_eval import partial_eval
+from chaste_codegen.chaste_model import ChasteModel
+
+
+class RlModel(ChasteModel):
+    """ Holds template and information specific for the RushLarsen model type"""
+
+    def __init__(self, model, file_name, **kwargs):
+        super().__init__(model, file_name, **kwargs)
+        self._derivative_equations = \
+            partial_eval(self._derivative_equations, self._y_derivatives, keep_multiple_usages=False)
+        self._non_linear_state_vars = \
+            get_non_linear_state_vars(self._derivative_equations, self._membrane_voltage_var,
+                                      self._state_vars, self._printer)
+
+        self._derivative_alpha_beta, self._formatted_alpha_beta_eqs, self._vars_in_derivative_alpha_beta = \
+            self._get_formatted_alpha_beta()
+        self._formatted_alpha_beta_eqs = self._format_alpha_beta_eqs()
+
+    def _get_formatted_alpha_beta(self):
+        """Gets the information for rAlphaOrTau, rBetaOrInf and formatted equations"""
```

Comment:
The names are probably different here - rAlphaOrTau looks like C++ naming convention?